<?xml version="1.0"?>

<launch>
    <arg name="rviz" default="true"/>
    <arg name="bhand" default="false"/>

    <param name="gripper" value="$(arg bhand)"/>
    <param name="/bullet_ros/joint_names" value=""/>
    <param name="/bullet_ros/init_pos" value=""/>
    <param name="robot_name" value="iiwa"/>


    <param unless="$(arg bhand)" name="robot_description" textfile="$(find bullet_ros_kuka)/urdf/kuka.urdf"/>
    <param unless="$(arg bhand)" name="/iiwa/robot_description" textfile="$(find bullet_ros_kuka)/urdf/kuka.urdf"/>

    <param if="$(arg bhand)" name="robot_description" textfile="$(find bullet_ros_kuka)/urdf/kuka_bhand.urdf"/>
    <param if="$(arg bhand)" name="/iiwa/robot_description" textfile="$(find bullet_ros_kuka)/urdf/kuka_bhand.urdf"/>

    <include file="$(find iiwa_bhand_moveit_config)/launch/moveit.launch">
        <arg name="rviz" value="$(arg rviz)"/>
        <arg name="bhand" value="$(arg bhand)"/>
        <arg name="sim" value="true"/>
    </include>

          <!-- Load controller settings -->
    <group ns="iiwa">
        <rosparam file="$(find bullet_ros_kuka)/config/iiwa_controllers.yaml" command="load"/>
<!--        <remap to="/PositionJointInterface_trajectory_controller/follow_joint_trajectory" from="/iiwa/PositionJointInterface_trajectory_controller/follow_joint_trajectory"/>-->
        <remap to="/iiwa/command/JointVelocity" from="/iiwa/joint_group_vel_controller/command"/>
        <arg name="controllers" default="joint_state_controller PositionJointInterface_trajectory_controller"/>

        <node name="controller_spawner_main" pkg="controller_manager" type="spawner" respawn="false" output="screen"
              args="$(arg controllers)"/>
        <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
              args="--stopped joint_group_vel_controller"/>

        <node name="bullet_ros_kuka" type="bullet_ros_kuka" pkg="bullet_ros_kuka" output="screen"/>
    </group>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />
    <node name="pyKUKA" pkg="bullet_ros_kuka" type="main.py" output="screen" args="bhand"/>
<!--     <node name="ext_torque_node" pkg="bullet_ros_kuka" type="ext_torque_node.py" output="screen"/> -->

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" respawn="true" output="screen">
            <rosparam param="source_list">["/iiwa/joint_states", "/bhand_node/joint_states"]</rosparam>
            <rosparam param="rate">250</rosparam>
    </node>
</launch>
